### PR TITLE
add internal/db/rdb/mysql/option test

### PR DESCRIPTION
**Pull request by vankichi**
Signed-off-by: vankichi <kyukawa315@gmail.com>

### Description:

- create test for `internal/db/rdb/mysql/option`
- update comment for each function in `internal/db/rdb/mysql/option`

### Environment:

- Go Version: 1.14.4
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.0

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [x] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/db/rdb/mysql/option_test.go [hunk 1]
```diff
@@ -19,1704 +19,1131 @@ package mysql
...
+				lim: "10s",
+			},
+			want: want{
+				obj: &T{
+					initialPingTimeLimit: 10 * time.Second,
+				},
+			},
+		},
+		{
+			name: "set success when lim is invalid",
```
Comment (kevindiu) [inline internal/db/rdb/mysql/option_test.go:650]:
```suggestion
			name: "set success with default value when lim is invalid",
```

File: internal/db/rdb/mysql/option_test.go [hunk 2]
```diff
@@ -19,1704 +19,1131 @@ package mysql
...
+				dur: "100ms",
+			},
+			want: want{
+				obj: &T{
+					initialPingDuration: 100 * time.Millisecond,
+				},
+			},
+		},
+		{
+			name: "set success when dur is invalid",
```
Comment (kevindiu) [inline internal/db/rdb/mysql/option_test.go:732]:
```suggestion
			name: "set success with default value when dur is invalid",
```

File: internal/db/rdb/mysql/option_test.go [hunk 3]
```diff
@@ -19,1704 +19,1131 @@ package mysql
...
+				dur: "60s",
+			},
+			want: want{
+				obj: &T{
+					connMaxLifeTime: 60 * time.Second,
+				},
+			},
+		},
+		{
+			name: "set success when dur is invalid",
```
Comment (kevindiu) [inline internal/db/rdb/mysql/option_test.go:814]:
```suggestion
			name: "set success with default value when dur is invalid",
```

File: internal/db/rdb/mysql/option_test.go [hunk 4]
```diff
@@ -19,1704 +19,1131 @@ package mysql
...
+			}
+		}(),
+		func() test {
+			return test{
+				name: "set success when cfg is nil",
+				want: want{
+					obj: new(T),
+				},
+			}
+		}(),
```
Comment (kevindiu) [inline internal/db/rdb/mysql/option_test.go:1047]:
```suggestion
		{
			name: "set success when cfg is nil",
			want: want{
				obj: new(T),
			},
		},
```

File: internal/db/rdb/mysql/option_test.go [hunk 5]
```diff
@@ -19,1704 +19,1131 @@ package mysql
...
+		func() test {
+			return test{
+				name: "set success when der is nil",
+				want: want{
+					obj: new(T),
+				},
+			}
+		}(),
```
Comment (kevindiu) [inline internal/db/rdb/mysql/option_test.go:1125]:
```suggestion
		{
			name: "set success when der is nil",
			want: want{
				obj: new(T),
			},
		},
```

File: internal/db/rdb/mysql/option_test.go (hunk 4 above)
Comment (vankichi) [reply internal/db/rdb/mysql/option_test.go:1047]: I'd not like to contain some different formats for readability 😕 
How do you think? I discussed about this before with @hlts2

Comment (kevindiu) [reply internal/db/rdb/mysql/option_test.go:1047]: @vankichi I think my suggestion is easier to read (the default format of table driven test) so I suggested here. 
Both fine for me :)
@hlt2 what do you think?

Comment (hlts2) [reply internal/db/rdb/mysql/option_test.go:1047]: @vankichi @kevindiu 
I discussed about this before with @vankichi 
I agreed to unify the formats.
And I recently thought that the unification improved the readability.

Comment (kevindiu) [reply internal/db/rdb/mysql/option_test.go:1047]: both fine for me :) will resolve the conversation

**Pull request merged**